pkg/peermgr/piermgr.go

```diff
@@ -0,0 +1,228 @@
+package peermgr
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/meshplus/bitxhub-model/pb"
+)
+
+const (
+	askTimeout = 10 * time.Second
+)
+
+func (swarm *Swarm) Piers() *Piers {
+	return swarm.piers
+}
+
+func (swarm *Swarm) AskPierMaster(address string) (bool, error) {
+	message := &pb.Message{
+		Data: []byte(address),
+		Type: pb.Message_CHECK_MASTER_PIER,
+	}
+
+	if swarm.piers.pierChan.checkAddress(address) {
+		return false, fmt.Errorf("is checking pier master")
+	}
+
+	ch := swarm.piers.pierChan.newChan(address)
+
+	for id := range swarm.Peers() {
```

Comment:
is swarm.Peers() contains self peer?

Reply:
it contains self peer, but it won't async send message to self-peer